providers/fips/self_test.c

```diff
@@ -216,9 +217,10 @@ static int verify_integrity(OSSL_CORE_BIO *bio, OSSL_FUNC_BIO_read_ex_fn read_ex
 
 static void set_fips_state(int state)
 {
-    CRYPTO_THREAD_write_lock(fips_state_lock);
-    FIPS_state = state;
-    CRYPTO_THREAD_unlock(fips_state_lock);
+    if (ossl_assert(CRYPTO_THREAD_write_lock(fips_state_lock) != 0)) {
```

Comment:
Doesn't CRYPTO_THREAD_write_lock() use ossl_assert already?

Reply:
No. At least not in the pthreads implementation.

Reply:
Well..at least there are some codepaths that call assert on failure but not others.

Reply:
I think it makes sense to put ossl_assert into the locking calls.  What do you think?

Reply:
The question is whether they could ever fail for reasons other than a programming error. If they can then assert is probably not appropriate. If not then it probably is. If the only reason that a locking call will ever fail is because of a programming error then I see no reason not to put the asserts in the locking calls themselves.

Reply:
locking can fail because of lack of system resources. It is unlikely, but possible, just like malloc failing on other than embedded systems.

Reply:
Should it raise an error?

Reply:
> should it raise an error?

I am going to say "not in this PR"  There is already #14321 to capture some future work thoughts.